src/test/java/io/jenkins/plugins/analysis/warnings/recorder/pageobj/SourceCodeView.java

```diff
@@ -0,0 +1,133 @@
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+/**
+ * Page object for the source code view page.
+ */
+public class SourceCodeView {
+    /** The html element id of the main panel. */
+    public static final String MAIN_PANEL_ELEMENT_ID = "main-panel";
+
+    /** The regex pattern describing the head line of the page (including file name). */
+    public static final Pattern FILE_NAME_PATTERN = Pattern.compile("Content of file (.*)");
+
+    /** The xPath to the file name (starting at the main panel). */
+    public static final String FILE_NAME_XPATH = "//h1";
+    /** The xPath to the source code (starting at the main panel). */
+    public static final String SOURCE_CODE_XPATH = "//pre//code";
+    /** The xPath to the source code producing the issue (starting at the main panel). */
+    public static final String ISSUE_SOURCE_CODE_XPATH = "//pre//code[2]";
+    /** The xPath to the message describing the issue (starting at the main panel). */
+    public static final String ISSUE_MESSAGE_XPATH = "//pre//div//div";
+
+    /** The source code view html page. */
+    private HtmlPage sourceCodeViewPage;
+
+    /**
+     * Constructor.
+     *
+     * @param sourceCodeViewPage
+     *         The source code view html page.
+     */
+    public SourceCodeView(HtmlPage sourceCodeViewPage) {
+        this.sourceCodeViewPage = sourceCodeViewPage;
+    }
+
+    /**
+     * Extracts the name of the file that is displayed on the html page.
+     *
+     * @return The file name.
+     */
+    public String getFileName() {
+        String name;
+
+        List<DomNode> headingByXPath = getMainPanel().getByXPath(FILE_NAME_XPATH);
+        if (!headingByXPath.isEmpty()) {
+            String heading = headingByXPath.get(0).asText();
+            Matcher matcher = FILE_NAME_PATTERN.matcher(heading);
+            if (matcher.matches()) {
+                return matcher.group(1);
+            }
+        }
+
+        throw new NoSuchElementException("Can't find the file name on page.");
+    }
+
+    /**
+     * Extracts the whole source code displayed on the html page.
+     *
+     * @return The source code.
+     */
+    public String getSourceCode() {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<DomNode> sourceCodeList = getMainPanel().getByXPath(SOURCE_CODE_XPATH);
+
+        for (DomNode node : sourceCodeList) {
+            String code = node.asText();
+            stringBuilder.append(code);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Extracts the source code line that produces the issue.
+     *
+     * @return The source code line producing the issue.
+     */
+    public String getIssueSourceCodeLine() {
```

Comment:
These might be multiple lines. So `getAffectedLines` might be a better name.

Reply:
We are not 100% procent sure if this method is actually necessary. Especially since the formatting is very hard to extract out of the HTML page. 
Is it fine if we remove it?